### ISSUE TITLE
"Percent Containing…" aggregation option

### ISSUE THREAD

**Issue by mazameli** [metabase/metabase#2300]
It would be useful if the query builder had an aggregation option called "Percent Containing…", which would let me choose a field and input a string or number that I want to match against. For example, if I had a field called `Venues`, and one of the values each venue can contain is `booked`, I could use this aggregation to find out what percent of my venues are booked.

**Comment by tlrobinson** [comment]
It would be nice if we could just use the existing aggregations and filter functionality rather than adding a new type of aggregation that needs a filter, which would also have the advantage of being more general purpose. What if you want the percent of the sum of a column rather than percent of the count or rows? Or what if you want to filter by a boolean, number, or date operator rather than a string?

I think this could be done with virtual tables that start from an existing question, plus a "percent of total" aggregation / calculated column?

**Comment by Caerbannog** [comment]
As a workaround you can use queries like these:
```sql
SELECT
    (100.0 * SUM(CASE WHEN "firstName" IS NOT NULL THEN 1 END) / COUNT(1)) AS firstNameFillingRate,
    (100.0 * SUM(CASE WHEN "age" < 18 THEN 1 END) / COUNT(1)) AS minorPercent,
FROM "public"."Users"
```

**Comment by sbelak** [comment]
Closed in favour of #9431
